### PR TITLE
Migration docs

### PR DESCRIPTION
**Pull request by jtmal-signalfx**

### CODE REVIEW

File: MIGRATING.md [hunk 1]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
+
+1. `SIGNALFX_SERVICE_NAME` or `service_name` configuration parameter
+    - to set in environment use `OTEL_SERVICE_NAME`
+    - to set in code use `Splunk::Otel.configure(service_name: 'your service name', ...`
+
+1. `SIGNALFX_ACCESS_TOKEN` or `access_token` configuration parameter
+    - can only be set in environment using `SPLUNK_ACCESS_TOKEN`
+
+1. `SIGNALFX_SPAN_TAGS` or `span_tags`
+    - TODO: check if we provide a way set resource
```
Comment (jtmal-signalfx) [inline MIGRATING.md:126]: I'll be looking into this next, I suggest that we just keep this TODO here until I get it resolved

**Comment by jtmal-signalfx** [comment]
@aurbiztondo-splunk Let's focus mostly on the language for now, get this into good-enough shape for beta partners, and we'll keep improving as we migrate customers.

File: MIGRATING.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Migrate from the SignalFx Tracing Library for Ruby
 
-The SignalFx Tracing Library for Ruby is deprecated. Replace it with the
+The SignalFx Tracing Library for Ruby will soon be deprecated. Replace it with the
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:3]: I think this is too vague. Do we have any dates for this? Can we say something like "will be deprecated by the end of 2022", for instance?

File: MIGRATING.md [hunk 3]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
+
+We're currently only testing installation using Bundler with RubyGems as a source.
+[Contact us](mailto:ssg-observability-instrumentals-ruby@splunk.com) if you require a different way of installation.
+
+If you installed
+[Smart Agent](https://github.com/signalfx/signalfx-agent)
+to serve as a gateway (traces/metrics data proxy) then,
+if possible, migrate to
+[OpenTelemetry Collector](https://docs.splunk.com/Observability/gdi/opentelemetry/resources.html)
+first.
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:67]: Suggestion: If you installed [Smart Agent](https://github.com/signalfx/signalfx-agent)
to serve as a gateway (traces/metrics data proxy), migrate to
[OpenTelemetry Collector](https://docs.splunk.com/Observability/gdi/opentelemetry/resources.html)
first, as soon as (if) possible.

File: MIGRATING.md [hunk 4]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
 
 To install the Splunk Distribution of OpenTelemetry Ruby, see the [README.md](README.md).
 
+### Replace SignalFx / OpenTracing gems with OTel equivalents
+
+1.  Make a list of all `signalfx-*` instrumentation gems from your Gemfile, see
+    [this table](https://github.com/signalfx/signalfx-ruby-tracing#supported-libraries)
+    for a complete list.
+
+1.  Replace them with Open Telemetry equivalents per the table below.
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:79]: Numbering.

Comment (aurbiztondo-splunk) [reply MIGRATING.md:79]: Replace them with Open Telemetry equivalents, as per the table below.

File: MIGRATING.md [hunk 5]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
-variables:
+1. `SIGNALFX_ENDPOINT_URL` or `ingest_url` configuration parameter
+    - if you installed Smart Agent and can migrate to OpenTelemetry Collector, do that first, then see the point below
+    - if you have OpenTelemetry Collector available up as a sidecar (via `localhost`),
+      and it accepts OTLP on default ports, then just remove this setting, we export OTLP to OTel Collector by default
+    - if you need to keep using Smart Agent, then you need to set up a jaeger exporter yourself
+    - if you export directly to our backend (without OTel Collector as a proxy),
+      then just set `SPLUNK_REALM` to your realm
+
+1. `SIGNALFX_SERVICE_NAME` or `service_name` configuration parameter
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:118]: Numbering

File: MIGRATING.md [hunk 6]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
+    - to set in environment use `OTEL_SERVICE_NAME`
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:119]: to set in environment, use `OTEL_SERVICE_NAME`

File: MIGRATING.md [hunk 7]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
+    - to set in code use `Splunk::Otel.configure(service_name: 'your service name', ...`
+
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:121]: to set in code, use `Splunk::Otel.configure(service_name: 'your service name', ...`

File: MIGRATING.md [hunk 8]
```diff
@@ -36,26 +48,88 @@ Follow these steps to remove the tracing library and its dependencies:
...
+1. `SIGNALFX_ACCESS_TOKEN` or `access_token` configuration parameter
+    - can only be set in environment using `SPLUNK_ACCESS_TOKEN`
+
+1. `SIGNALFX_SPAN_TAGS` or `span_tags`
+    - TODO: check if we provide a way set resource
+
+1. `SIGNALFX_RECORDED_VALUE_MAX_LENGTH`
+    - can be set in environment using `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT`
+    - see <https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#span-limits>
```
Comment (aurbiztondo-splunk) [inline MIGRATING.md:130]: Add a description for the url

**Review by aurbiztondo-splunk — APPROVED** [review]
Left a few comments!

File: MIGRATING.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
 # Migrate from the SignalFx Tracing Library for Ruby
 
-The SignalFx Tracing Library for Ruby is deprecated. Replace it with the
-agent from the Splunk Distribution of OpenTelemetry Ruby.
+The SignalFx Tracing Library for Ruby is scheduled to be deprecated by the end of 2022.
+Replace it with the agent from the Splunk Distribution of OpenTelemetry Ruby.
```
Comment (tsloughter-splunk) [inline MIGRATING.md:4]: I don't think we should use the term "agent" here and in other places in this document that aren't part of this PR. Just call it the Distribution or a library?

File: MIGRATING.md [hunk 10]
```diff
@@ -36,26 +48,109 @@ Follow these steps to remove the tracing library and its dependencies:
...
 To install the Splunk Distribution of OpenTelemetry Ruby, see the [README.md](README.md).
 
-### Migrate settings for the Splunk Ruby OTel agent
+### Replace SignalFx / OpenTracing gems with OTel equivalents
+
+1.  Make a list of all `signalfx-*` instrumentation gems from your Gemfile, see
+    [this table](https://github.com/signalfx/signalfx-ruby-tracing#supported-libraries)
+    for a complete list.
+
+1.  Replace them with Open Telemetry, as per the table below.
```
Comment (tsloughter-splunk) [inline MIGRATING.md:79]:
```suggestion
1.  Replace them with OpenTelemetry, as per the table below.
```

File: MIGRATING.md [hunk 11]
```diff
@@ -36,26 +48,109 @@ Follow these steps to remove the tracing library and its dependencies:
...
+
+1.  Replace any other OpenTracing instrumentation packages you might have installed yourself.
```
Comment (tsloughter-splunk) [inline MIGRATING.md:81]:
```suggestion
2.  Replace any other OpenTracing instrumentation packages you might have installed yourself.
```

File: MIGRATING.md [hunk 12]
```diff
@@ -15,6 +15,17 @@ Ruby OTel agent.
...
+
+- Different subset of supported Ruby versions, see previous section
+- No auto-instrumentation for
+  (more details in
+  [OTel instrumentation equivalents](#signalfx-instrumentations-equivalents)
+  ):
+    - `elasticsearch` (<https://github.com/elastic/elasticsearch-ruby>)
+    - `grape` (<https://github.com/ruby-grape/grape>)
+    - `sequel` (<https://github.com/jeremyevans/sequel>)
+
```
Comment (tsloughter-splunk) [inline MIGRATING.md:28]:
```suggestion
- Currently no auto-instrumentation for:
    - `elasticsearch` (<https://github.com/elastic/elasticsearch-ruby>)
    - `grape` (<https://github.com/ruby-grape/grape>)
    - `sequel` (<https://github.com/jeremyevans/sequel>)

More details in [OTel instrumentation equivalents](#signalfx-instrumentations-equivalents).
```

File: MIGRATING.md [hunk 13]
```diff
@@ -36,26 +48,109 @@ Follow these steps to remove the tracing library and its dependencies:
     gem uninstall signalfx
     ```
 
-2.  Remove `signalfx` from your Gemfile.
+1.  Remove `signalfx` from your Gemfile.
```
Comment (tsloughter-splunk) [inline MIGRATING.md:51]:
```suggestion
2.  Remove `signalfx` from your Gemfile.
```

File: MIGRATING.md [hunk 14]
```diff
@@ -36,26 +48,109 @@ Follow these steps to remove the tracing library and its dependencies:
...
+           end
+           ```
+
+1. `SIGNALFX_RECORDED_VALUE_MAX_LENGTH`
+    - can be set in environment using `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT`
+    - see
+      [the list of span limits settings](<https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#span-limits>)
+      to learn more
+
+1. `tracer` configuration parameter
```
Comment (tsloughter-splunk) [inline MIGRATING.md:153]:
```suggestion
2. `tracer` configuration parameter
```

**Review by tsloughter-splunk — APPROVED** [review]

**Pull request merged**